pkg/kubectl/cmd/expose.go

```diff
@@ -175,6 +176,9 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		}
 		params["labels"] = kubectl.MakeLabels(labels)
 	}
+	if cmdutil.GetFlagBool(cmd, "headless") {
```

Comment:
Also you probably need etcd in your PATH. If you already point GOPATH/bin to it, then `go get github.com/coreos/etcd` and you are set.